platforms/ide/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiDistributionResolver.groovy

```diff
@@ -16,102 +16,93 @@
 
 package org.gradle.integtests.tooling.fixture
 
-import org.gradle.api.artifacts.Configuration
-import org.gradle.api.artifacts.Dependency
-import org.gradle.api.internal.artifacts.DependencyResolutionServices
-import org.gradle.api.internal.project.ProjectInternal
 import org.gradle.integtests.fixtures.RepoScriptBlockUtil
 import org.gradle.integtests.fixtures.executer.CommitDistribution
 import org.gradle.integtests.fixtures.executer.IntegrationTestBuildContext
+import org.gradle.internal.classloader.ClasspathUtil
 import org.gradle.internal.exceptions.DefaultMultiCauseException
-import org.gradle.testfixtures.ProjectBuilder
-import org.gradle.testfixtures.internal.ProjectBuilderImpl
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 
-class ToolingApiDistributionResolver {
-
-    static interface ResolverAction<T> {
-        T run(ToolingApiDistributionResolver resolver)
-    }
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.StandardCopyOption
+import java.util.function.Supplier
 
-    /**
-     * Executes given {@code block} against a fresh instance of the {@code ToolingApiDistributionResolver}
-     * and returns the result.
-     */
-    static <T> T use(ResolverAction<T> block) {
-        def project = (ProjectInternal) ProjectBuilder.builder().build()
-        try {
-            def resolver = new ToolingApiDistributionResolver(project)
-            return block.run(resolver)
-        } finally {
-            ProjectBuilderImpl.stop(project)
-        }
-    }
+class ToolingApiDistributionResolver {
```

Comment:
Can you add a test that uses this fixture to download a couple of versions and checks the behavior:
- We download them to the expected location
- We don't download them twice
- We produce an OK error if we ask for something that doesn't exist/the download fails

To check if we download it twice, you can just set the modified time on the file to something in the past and check if it changes.  You don't need to standup a full HTTP server. 

I know this is exercised in the actually cross-version tests, but this will be easier to check if we want to make improvements.